src/evaluatorfrontend.cpp

```diff
@@ -67,18 +74,82 @@ void EvaluatorFrontend::FuncRunner() {
 		Plot();
 }
 
-//! @file
-// [runeval]
-void EvaluatorFrontend::RunEvaluator() {
-	while (!evaluator->IsFinished()) {
+int EvaluatorFrontend::EvaluatorFunc() {
+	if (!evaluator->IsFinished()) {
 		states.push_back(evaluator->GetNextNetworkState());
 		if (printStd) {
 			std::cout << states.back().PrintCsvRow();
 		}
+	} else {
+		return 0;
 	}
+
+	return 1;
+}
+
+//! @file
+// [runeval]
+void EvaluatorFrontend::RunEvaluator() {
+	while (EvaluatorFunc())
+		;
 }
 // [runeval]
 
+void EvaluatorFrontend::RunRTEvaluator() {
+	auto StartT = std::chrono::steady_clock::now();
+	auto EndT = std::chrono::steady_clock::now();
+	bool once = true;
+
+	while (!mPlot->RunPlot()) {
+		if (EvaluatorFunc()) {
+			EndT = std::chrono::steady_clock::now();
+			auto ETime =
+					std::chrono::duration_cast<std::chrono::milliseconds>(EndT - StartT);
+			auto UTime = std::chrono::milliseconds(UpdateRate);
+
+			if (ETime > UTime) {
+				StartT = std::chrono::steady_clock::now();
+
+				for (int i = 0; i < ToPlot.size(); i++) {
+					for (int j = LastInsertedCount; j < states.size(); j++) {
+						auto state = states[j];
+						ToPlot[i].Function.push_back((OpenRTP::Point){
+								(float)state.time, (float)state[ToPlot[i].Name]});
```

Comment:
C++ style `static_cast<float>(state.time)` are prefered to C-style explicit casts. The syntax is clearer, and C++ style casts are checked at compile time, while C-style casts are checked at runtime